arm-machinelearning/2016-05-01-preview/swagger/commitmentPlans.json

```diff
@@ -0,0 +1,720 @@
+      "required": true
+    },
+    "APIVersionParameter": {
+      "in": "query",
+      "name": "api-version",
+      "type": "string",
+      "description": "The versiong of the Microsoft.MachineLearning resource provider API to be used.",
+      "required": true
+    }
+  },
+  "definitions": {
+    "Resource": {
+      "x-ms-azure-resource": true,
+      "description": "Common properties of an ARM resource.",
+      "required": [
+        "location"
+      ],
+      "properties": {
+        "id": {
+          "description": "Resource Id",
+          "type": "string",
+          "readOnly": true
+        },
+        "name": {
+          "description": "Resource Name",
+          "type": "string"
+        },
+        "location": {
+          "description": "Resource Location",
+          "type": "string"
+        },
+        "type": {
+          "description": "Resource type",
+          "type": "string",
+          "readOnly": true
+        },
+        "tags": {
+          "description": "Resource tags",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "CommitmentAssociation": {
+      "type": "object",
+      "description": "Represents the association between a commitment plan and some other resource, such as a Machine Learning web service.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "properties": {
+        "etag": {
+          "type": "string",
+          "description": "An entity tag used to enforce optimistic concurrency."
+        },
+        "properties": {
+          "$ref": "#/definitions/CommitmentAssociationProperties"
+        }
+      }
+    },
+    "CommitmentAssociationProperties": {
+      "type": "object",
+      "description": "Properties of an Azure ML commitment association.",
+      "properties": {
+        "associatedResourceId": {
+          "type": "string",
+          "description": "The ID of the resource this association points to, such as the ARM ID of an Azure ML web service."
+        },
+        "commitmentPlanId": {
+          "type": "string",
+          "description": "The ARM ID of the parent Azure ML commitment plan."
+        },
+        "creationDate": {
+          "format": "date-time",
+          "type": "string",
+          "description": "The date at which this commitment association was created, in ISO 8601 format."
+        }
+      }
+    },
+    "ResourceSku": {
+      "type": "object",
+      "description": "The SKU of a resource.",
+      "properties": {
+        "capacity": {
+          "format": "int32",
+          "type": "integer",
+          "description": "The scale-out capacity of the resource. 1 is 1x, 2 is 2x, etc. This impacts the quantities and cost of any commitment plan resource."
+        },
+        "name": {
+          "type": "string",
+          "description": "The SKU name. Along with tier, uniquely identifies the SKU."
```

Comment:
It's not possible to enumerate SKUs because they are changed independently of any client API and are not the same for all callers. ARM provides a SKUs API that clients use to discover the possible values, which are also filtered by location and properties of the subscription.